amacc.c

```diff
@@ -21,19 +21,19 @@ char **scnames; // system call names
 int *e, *le, *text;  // current position in emitted code
 int *cas;            // case statement patch-up pointer
 int *brks;           // break statement patch-up pointer
-int *def;            // default statement patch-up pointer
 int *tsize;          // array (indexed by type) of type sizes
 int tnew;            // next available type
 int tk;              // current token
 int ival;            // current token value
 int ty;              // current expression type
-int loc;             // local variable offset
 int line;            // current line number
 int src;             // print source and assembly flag
 int signed_char;     // use `signed char` for `char`
 int elf;             // print ELF format
 int elf_fd;
 int rwdata_align_off;
+int* n;
```

Comment:
The naming `n` is too general to distinguish.

Reply:
Will all `e` be replaced by `n`?

Reply:
No, the e is for emit IR and the n is for emit ast , they are different.

Reply:
Add comments as following:
> With an AST (abstracted syntax tree), the compiler is not limited to generating code on the fly with parsing.
> This capability allows function parameter code to be emitted and pushed on the stack in the proper right-to-left order.